src/Containers/VendorsList/NavigationItem.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import styled from 'styled-components';
+import { StyledIcon } from '@styled-icons/styled-icon';
+import { Heading, HeadingProps } from '../../Text/Heading';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+import { flex } from '../../Utils/Mixins';
+
+export interface INavigationItemProps extends MainInterface, ResponsiveInterface, React.HTMLAttributes<HTMLDivElement> {
+    icon?: StyledIcon,
+    label: string,
+    selectedItem: string,
+    headingProps?: HeadingProps,
+    iconProps?: IIconProps,
+};
+
+export const NavigationItem: React.FC<INavigationItemProps> = ({
+    icon,
+    label,
+    selectedItem,
+    headingProps,
+    iconProps,
+    ...props
+}): React.ReactElement => {
+    return (
+        <Wrapper isSelected={selectedItem === label} {...props}>
+            <Icon as={icon} {...iconProps} />
+            <Heading type='h6' {...headingProps}>
+                {label}
+            </Heading>
+        </Wrapper>
+    );
+};
+
+interface IWrapperProps{
+    isSelected: boolean
+};
+const Wrapper = styled.div<IWrapperProps>`
```

Comment:
You could also do Pick<INavigationItemProps, 'selectedItem'> then pass selectedItem down and do the comparison in the Styled Component as well. That would keep the types in sync / reduce extra code being written

Reply:
Love to hear your comment on this though ^

Reply:
Interesting, never thought of this method, but I do agree that it is cleaner / saves lines of code. I have to pass in two props though since it relies on a boolean value? Not sure if this still works well